.github/workflows/check-full.yaml

```diff
@@ -12,31 +12,44 @@ jobs:
   R-CMD-check:
     runs-on: ${{ matrix.config.os }}
 
-    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }}) ${{ matrix.config.locale }}
 
     strategy:
       fail-fast: false
       matrix:
         config:
-          - {os: macos-latest,   r: 'release'}
+          - { os: macos-latest, r: "release" }
 
-          - {os: windows-latest, r: 'devel'}
-          - {os: windows-latest, r: 'release'}
+          - { os: windows-latest, r: "devel" }
+          - { os: windows-latest, r: "release" }
           # use 4.1 to check with rtools40's older compiler
-          - {os: windows-latest, r: '4.1'}
+          - { os: windows-latest, r: "4.1" }
 
-          - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
-          - {os: ubuntu-latest,   r: 'release'}
-          - {os: ubuntu-latest,   r: 'oldrel-1'}
-          - {os: ubuntu-latest,   r: 'oldrel-2'}
-          - {os: ubuntu-latest,   r: 'oldrel-3'}
-          - {os: ubuntu-latest,   r: 'oldrel-4'}
+          - { os: ubuntu-latest, r: "devel", http-user-agent: "release" }
+          - { os: ubuntu-latest, r: "devel", locale: "en_US" }
+          #- { os: ubuntu-latest, r: "release", locale: "zh_CN" }
+          - { os: ubuntu-latest, r: "oldrel-1" }
+          - { os: ubuntu-latest, r: "oldrel-2" }
+          - { os: ubuntu-latest, r: "oldrel-3" }
+          - { os: ubuntu-latest, r: "oldrel-4" }
 
     env:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
       R_KEEP_PKG_SOURCE: yes
 
     steps:
+      - name: Set locale
```

Comment:
Can we have just one step for setting the locale and reference the value from the matrix as an env variable? If not directly, turn the context into a env variable first and then access it? Like this? 
```yaml
      - name: Set locale
        env:
          OVERRIDE_LOCALE: '${{ matrix.config.locale }}'
        run: echo $OVERRIDE_LOCALE >> $GITHUB_ENV

```

Reply:
The solutions to DRYing this up was leading to more complicate code, so I am going to retain the more readable individual steps for now.